### PR TITLE
Deprecate passing options to Gem::GemRunner

### PR DESCRIPTION
**Pull request by maclover7**
# Description:

Would use `Gem::Deprecate` but we're not deprecating a method here 😢 
______________

# Tasks:

- [ ] Describe the problem / feature
- [ ] Write tests
- [ ] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: lib/rubygems/gem_runner.rb [hunk 1]
```diff
@@ -26,7 +26,10 @@
 class Gem::GemRunner
 
   def initialize(options={})
-    # TODO: nuke these options
+    if options[:command_manager] || options[:config_file]
```
Comment (segiddins) [inline lib/rubygems/gem_runner.rb]: Maybe just check `!options.empty?`

File: lib/rubygems/gem_runner.rb [hunk 2]
```diff
@@ -26,7 +26,10 @@
...
+      Kernel.warn "NOTE: passing options to Gem::GemRunner.new is deprecated with no replacement. It will be removed on or after 2016-10-01."
```
Comment (segiddins) [inline lib/rubygems/gem_runner.rb]: Needs to check for `Gem::Deprecate.skip`

**Comment by maclover7** [comment]
updated @segiddins

**Comment by segiddins** [comment]
@homu r+

File: lib/rubygems/gem_runner.rb [hunk 3]
```diff
@@ -26,7 +27,10 @@
 class Gem::GemRunner
 
   def initialize(options={})
-    # TODO: nuke these options
+    if !options.empty? && !Gem::Deprecate.skip
+      Kernel.warn "NOTE: passing options to Gem::GemRunner.new is deprecated with no replacement. It will be removed on or after 2016-10-01."
```
Comment (segiddins) [inline lib/rubygems/gem_runner.rb]: Because this is the format `Gem.deprecate` uses

**Pull request merged**